Sources/NIOHTTP2/ConnectionStateMachine/ConnectionStateMachine.swift

```diff
@@ -557,46 +561,67 @@ extension HTTP2ConnectionStateMachine {
 
         switch self.state {
         case .idle(let state):
-            var settingsState = HTTP2SettingsState(localState: true)
-            settingsState.emitSettings(settings)
-            self.state = .prefaceSent(.init(fromIdle: state, localSettings: settingsState))
+            self.avoidingStateMachineCoW { [] newState in
```

Comment:
mind removing the left-over `[]` ?

Reply:
It's there on purpose to avoid me ever accidentally capturing in this block.

Reply:
@Lukasa that doesn't work. `[...]` has no effect on things that are not listed

Reply:
I complain for years that Swift needs a mode where I can specify an exhaustive capture list, but still not there :(

Reply:
the only thing `[` does is to create a by-value copy of all the listed things, optionally with renaming

Reply:
Fixed. :grin: